scipy/spatial/transform/_rigid_transform.pyx

```diff
@@ -90,22 +97,26 @@ cdef _normalize_dual_quaternion(real_part, dual_part):
     prod_real, prod_dual = _compose_dual_quaternion(
         real_part, dual_part, real_conjugate, dual_conjugate)
 
+    # the squared norm of a dual quaternion is a dual number s + t * epsilon
+    s = prod_real[:, 3]
+    t = prod_dual[:, 3]
+
+    prod_real_norm = np.sqrt(np.maximum(s, 0.0))
     # special case: invalid real quaternion
-    prod_real_norm = np.linalg.norm(prod_real, axis=1)
     invalid_real_mask = prod_real_norm == 0.0
     real_part[invalid_real_mask, :4] = [0., 0., 0., 1.]
     prod_real_norm[invalid_real_mask] = 1.0
 
-    # compute normalization factor
-    real_inv_sqrt = 1.0 / prod_real_norm
-    dual_inv_sqrt = -0.5 * prod_dual * real_inv_sqrt[:, np.newaxis] ** 3
+    # compute normalization factor: the inverse square root of s + t * epsilon
+    # is a dual number u + v * epsilon
+    u = 1.0 / prod_real_norm[:, np.newaxis]
+    v = -0.5 * t[:, np.newaxis] * u ** 3
 
     # normalize dual quaternion
-    real_part = real_inv_sqrt[:, np.newaxis] * real_part
-    dual_part = real_inv_sqrt[:, np.newaxis] * dual_part + np.asarray(
-        compose_quat(dual_inv_sqrt, real_part))
+    new_real_part = u * real_part
+    dual_part = u * dual_part + v * real_part
```

Comment:
Any intuition why these formulas apply? We multiply a dual number with a dual quaternion like 2 dual numbers basically, correct?

Reply:
Yes, the difficulty here is that we have to multiply the dual quaternion with the inverse square root of a dual number (in this case of the squared norm of a dual quat). I have the derivation on paper or you can find some (imo not so intuitive) version of it in the stack overflow answer that I linked at the top of the function. The point is, it's not straightforward to compute the inverse square root of a dual number. I can write it down tomorrow.